backend/src/helpers/functions.ts

```diff
@@ -0,0 +1,214 @@
+import {NextFunction, Response, Request} from "express";

```

Comment:
I think this file should be separated into multiple different files (now probably 2) and the folder they are in should be called something like `ViewModels`. 

As far as I know there are many different design patterns for BE application, but the most famous are MVC and MVVM and they all follow the same idea: 
- to have your interface (your api or you routes) in one place. Its basically the place where the interface for accessing your BE from outer world is defined. You have it all in `backend/index.ts`
- to have `Models` its where Entities are defined and probably linked to DB tables
- to have a place where your "business logic" is happening. In your case its this `functions.ts` file. 

You basically have it devided in a similar fashion, but its all a bit mixed up. 

Things I would suggest: 
- rename `helpers` folder into something that makes more sense. I would suggest `viewmodels` (MVVM). But maybe its not the best option, try to google MVVM and MVC patterns. Basically that is the folder where your business logic will be.
- inside this folder separate this `functions.ts` into `PostVM.ts` and `UserVM.ts` and move corresponding logic to this files. Idea is to keep Models and "business logic" logically grouped together. So you don't have one file responsible for everything, but you have 1 file with all logic for 1 model

Reply:
helpers function separate to controllers,
keep models, 
separate routes.